Test Test_SSHCreds_Environ_TempFileCleanupOnInvalidProxyURL fails on MacOS development environment

<!-- If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack). -->

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

<!-- A clear and concise description of what the bug is. -->

The test Test_SSHCreds_Environ_TempFileCleanupOnInvalidProxyURL fails on Macos development environment,
both with the local toolchain.

**To Reproduce**

<!-- A list of the steps required to reproduce the issue. Best of all, give us the URL to a repository that exhibits this issue. -->

Tested on recent macos (arm64 Sequoia 15.5).

1. clone Argo-Cd from its master branch, cd into the project directory
2. run `make install-tools-local`
3. run `make TEST_MODULE=github.com/argoproj/argo-cd/v3/util/git test-local` 
4. see the test Test_SSHCreds_Environ_TempFileCleanupOnInvalidProxyURL fail (see log below)

**Expected behavior**

All the tests should pass.

**Version**

```shell
argocd: v3.2.0+79f152c
  BuildDate: 2025-07-10T23:52:24Z
  GitCommit: 79f152c1ba98112f388cb3fe2df8c89c24f1d780
  GitTreeState: clean
  GoVersion: go1.24.4
  Compiler: gc
  Platform: darwin/arm64
```

**Logs**

```
 make TEST_MODULE=github.com/argoproj/argo-cd/v3/util/git test-local 
if test "github.com/argoproj/argo-cd/v3/util/git" = ""; then \
		DIST_DIR=/Users/dudin/go/src/github.com/dudinea/argo-cd/dist RERUN_FAILS=0 PACKAGES=`go list ./... | grep -v 'test/e2e'` ./hack/test.sh -args -test.gocoverdir="/Users/dudin/go/src/github.com/dudinea/argo-cd/test-results"; \
	else \
		DIST_DIR=/Users/dudin/go/src/github.com/dudinea/argo-cd/dist RERUN_FAILS=0 PACKAGES="github.com/argoproj/argo-cd/v3/util/git" ./hack/test.sh -args -test.gocoverdir="/Users/dudin/go/src/github.com/dudinea/argo-cd/test-results" "github.com/argoproj/argo-cd/v3/util/git"; \
	fi
+ which go-junit-report
/Users/dudin/go/bin/go-junit-report
+ TEST_RESULTS=test-results
+ TEST_FLAGS=
+ test '' '!=' ''
+ test '' '!=' ''
+ mkdir -p test-results
+ GODEBUG=tarinsecurepath=0,zipinsecurepath=0
+ gotestsum --rerun-fails-report=rerunreport.txt --junitfile=test-results/junit.xml --format=testname --rerun-fails=0 --packages=github.com/argoproj/argo-cd/v3/util/git -- -cover -args -test.gocoverdir=/Users/dudin/go/src/github.com/dudinea/argo-cd/test-results github.com/argoproj/argo-cd/v3/util/git

[SKIPPED]

=== Failed
=== FAIL: util/git Test_SSHCreds_Environ_TempFileCleanupOnInvalidProxyURL (0.00s)
    creds_test.go:334: 
        	Error Trace:	/Users/dudin/go/src/github.com/dudinea/argo-cd/util/git/creds_test.go:334
        	            				/Users/dudin/go/src/github.com/dudinea/argo-cd/util/git/creds_test.go:346
        	Error:      	Received unexpected error:
        	            	open : no such file or directory
        	Test:       	Test_SSHCreds_Environ_TempFileCleanupOnInvalidProxyURL

DONE 122 tests, 1 skipped, 1 failure in 43.441s
make: *** [test-local] Error 1
```

--- a/util/git/creds_test.go
+++ b/util/git/creds_test.go
@@ -328,9 +328,18 @@ func Test_SSHCreds_Environ_WithProxyUserNamePassword(t *testing.T) {
 func Test_SSHCreds_Environ_TempFileCleanupOnInvalidProxyURL(t *testing.T) {
 	// Previously, if the proxy URL was invalid, a temporary file would be left in /dev/shm. This ensures the file is cleaned up in this case.
 
-	// countDev returns the number of files in /dev/shm (argoutilio.TempDir)
+	// argoio.TempDir will be /dev/shm or "" (on an OS without /dev/shm).
+	// In this case os.CreateTemp(), which is used by creds.Environ(),
+	// will use os.TempDir for the temporary directory.
+	// Reproducing this logic here:
+	argoioTempDir := argoio.TempDir
+	if argoioTempDir == "" {
+		argoioTempDir = os.TempDir()
+	}
+
+	// countDev returns the number of files in the temporary directory
 	countFilesInDevShm := func() int {
-		entries, err := os.ReadDir(argoio.TempDir)
+		entries, err := os.ReadDir(argoioTempDir)
 		require.NoError(t, err)
 
 		return len(entries)

